pkg/controllermanager/controller/secretbinding/secretbinding.go

```diff
@@ -57,32 +56,29 @@ type Controller struct {
 func NewSecretBindingController(
 	ctx context.Context,
 	log logr.Logger,
-	clientMap clientmap.ClientMap,
-	recorder record.EventRecorder,
+	mgr manager.Manager,
 ) (
 	*Controller,
 	error,
 ) {
 	log = log.WithName(ControllerName)
 
-	gardenClient, err := clientMap.GetClient(ctx, keys.ForGarden())
-	if err != nil {
-		return nil, err
-	}
+	gardenClient := mgr.GetClient()
+	gardenCache := mgr.GetCache()
 
-	secretBindingInformer, err := gardenClient.Cache().GetInformer(ctx, &gardencorev1beta1.SecretBinding{})
+	secretBindingInformer, err := gardenCache.GetInformer(ctx, &gardencorev1beta1.SecretBinding{})
 	if err != nil {
 		return nil, fmt.Errorf("failed to get SecretBinding Informer: %w", err)
 	}
 
-	shootInformer, err := gardenClient.Cache().GetInformer(ctx, &gardencorev1beta1.Shoot{})
+	shootInformer, err := gardenCache.GetInformer(ctx, &gardencorev1beta1.Shoot{})
 	if err != nil {
 		return nil, fmt.Errorf("failed to get Shoot Informer: %w", err)
 	}
 
 	secretBindingController := &Controller{
-		reconciler:                      NewSecretBindingReconciler(gardenClient.Client(), recorder),
-		secretBindingProviderReconciler: NewSecretBindingProviderReconciler(gardenClient.Client()),
+		reconciler:                      NewSecretBindingReconciler(gardenClient, mgr.GetEventRecorderFor(controllermanager.Name)),
```

Comment:
Shouldn't we pass reconciler names here instead of the GCM name which is always the same? (same for other occurrences)

Reply:
Thought about this as well, but decided to go with `gardener-controller-manager` as this will pop up as the event source (i.e., very prominently in `kubectl describe`) and thought this would be clearer, where the events originate from.

Gardener components seem to mainly use the binary name, kubernetes components seem to mainly use the controller name.

We can go for the controller names if you want :)

Reply:
What do we do for the recorder in the shoot controller?
using `shoot-controller` is ambiguous 🙈

Reply:
Done as discussed.